packages/next/src/server/app-render/app-render.tsx

```diff
@@ -1505,6 +1507,17 @@ export async function renderToHTMLOrFlight(
           }
           if (isRedirectError(err)) {
             res.statusCode = 307
+            if (err.mutableCookies) {
+              const headers = getMutableCookieHeaders(
+                new Headers(),
```

Comment:
I'm not sure if headers could exist here. Do we have a function to convert legacy headers just in case?